azavea/raster-foundry
# Create ScenesToProjects with a default color correction

## Overview

Set default color correction values when adding a scene to a project
Add composites jsonb field to datasources
Use "natural" composite when creating default color correction if available,
default to the first one otherwise

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~

### Demo
![image](https://cloud.githubusercontent.com/assets/4392704/22835931/7ba74bc2-ef88-11e6-9459-689c704d3f80.png)

### Notes
The migration adds a default set of color modes for landsat to the datasources table. This will allow us to build out the UI to support it.

We do not have color modes defined for sentinel, so it will default to bands 1,2, and 3 for R,G, and B respectively. It might make sense to try and inspect the scene for available bands and set accordingly, or just not worry about it since we shouldn't allow ingesting scenes without a valid datasource (where valid means having composites etc defined)

## Testing Instructions

 * Run migrations
 * Create a new project, and add one of the ingested landsat scenes to it
 * Open the project editor using the created project
 * Verify that the project has a default color correction defined, with the natural color mode selected by default

Closes #1080 #1076
